### DISCUSSION TITLE
「Names must match /^[_a-zA-Z][_a-zA-Z0-9]*$/」 How to achieve name contains decimal point

### DISCUSSION THREAD

**Discussion by blueMountain007** [payloadcms/payload#842] [Q&A]
<img width="774" alt="image" src="https://user-images.githubusercontent.com/16400882/181433707-3818a29b-f542-4275-a49e-7316df9e6f4e.png">

**Comment by jmikrut** [comment]
Hey @zhounewz – GraphQL does not allow periods in names, so this is never going to be possible within GraphQL directly. But, we could further "sanitize" field names, replacing dots with double underscores or similar. What do you think about that? If it were me, I'd completely avoid using dot notation in field names because that's going to make your data a PIA to work with.

Consider the following: if you use camelcase, you can access your properties much more easily like `data.fieldName`. If you use periods in your field names, you would _always_ need to access them like `data['field.name']` which is an anti-pattern.

Thoughts?
